trustpinning/certs_test.go

```diff
@@ -363,6 +363,26 @@ func TestValidateRootWithPinnedCertAndIntermediates(t *testing.T) {
 		}
 	}
 	require.Equal(t, typedSignedRoot, validatedRoot)
+
+	// test is also works with a wildcarded gun in certs
+	validatedRoot, err = trustpinning.ValidateRoot(
+		nil,
+		signedRoot,
+		"docker.io/notary/test",
+		trustpinning.TrustPinConfig{
+			Certs: map[string][]string{
+				"docker.io/notar*": {ecdsax509Key.ID()},
+			},
+			DisableTOFU: true,
+		},
+	)
```

Comment:
can we add another test case or two for the precedence between globbed and non-globbed matches?

Ex:
this fails
```
Certs: map[string][]string{
    "docker.io/notary": {"badID"},
    "docker.io/notar*": {ecdsax509Key.ID()},
 },
```

this succeeds
```
Certs: map[string][]string{
    "docker.io/notary": {ecdsax509Key.ID()},
    "docker.io/notar*": {"badID"},
 },
```